### PR TITLE
Add keyboard polilla, a 60-key ergonomic board

### PR DESCRIPTION
**Pull request by elagil**
## Description

Added a new keyboard "polilla". This is a 60-key ergonomic board, made of layered PCBs: https://github.com/elagil/polilla

The board uses a crystal-less STM32F042K6T processor and supports VIA. For VIA support to be working (and therefore dynamic keymaps via emulated EEPROM), this PR is required as well: https://github.com/qmk/qmk_firmware/pull/10685

Is there a nicer way of integrating the ChibiOS config files or is there even a global include path for certain processor families? Adding them to the keyboard folder does not seem ideal.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/polilla/rev1/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,86 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/polilla/rev1/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/polilla/rev1/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,108 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/polilla/rev1/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/polilla/rev1/rules.mk [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+LTO_ENABLE = yes
```
Comment (drashna) [inline keyboards/polilla/rev1/rules.mk:14]: LTO has some weird behavior on ARM.  You may not want to have it enabled. 

```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/polilla/rev1/rules.mk (hunk 3 above)
Comment (elagil) [reply keyboards/polilla/rev1/rules.mk]: Thanks for your comments, I implemented your suggestions.

File: keyboards/polilla/rev1/config.h [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0x6166  // af
+#define PRODUCT_ID 0x0010
+#define DEVICE_VER 0x0001
+#define MANUFACTURER elagil
+#define PRODUCT Polilla
+#define DESCRIPTION Ergonomic keyboard with 60 keys
```
Comment (fauxpark) [inline keyboards/polilla/rev1/config.h:27]:
```suggestion
```

File: keyboards/polilla/rev1/config.h [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+#define STM32_BOOTLOADER_ADDRESS 0x1FFFC400
+
```
Comment (fauxpark) [inline keyboards/polilla/rev1/config.h:30]:
```suggestion
```

File: keyboards/polilla/rev1/rules.mk [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/polilla/rev1/rules.mk:13]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by elagil** [comment]
I removed the custom stack declaration. Therefore, this will rely on https://github.com/qmk/qmk_firmware/pull/10685 to be merged for the VIA keymap to compile.

**Comment by tzarc** [comment]
> Is there a nicer way of integrating the ChibiOS config files or is there even a global include path for certain processor families? Adding them to the keyboard folder does not seem ideal.

It's coming... part of the ChibiOS upgrade stuff in the pipeline is going to modify things such that the config files include overrides only -- this makes them ChibiOS version-agnostic.

For now, you can keep them as part of your PR -- I've mechanised the majority of the conversion (#10418) -- I'll add this board to the list when it's merged.

**Comment by elagil** [comment]
This is just got ready for merging, since https://github.com/qmk/qmk_firmware/pull/10685 is now part of QMK. :)

**Review by drashna — APPROVED** [review]

**Comment by elagil** [comment]
I wonder why the CI build fails. When I do it on the latest tag, it works fine. What could it be?

```
qmk_firmware$ make polilla
QMK Firmware 0.10.43
Making polilla/rev1 with keymap default                                                                [OK]
Making polilla/rev1 with keymap via                                                                    [OK]
```

**Comment by fauxpark** [comment]
Must be something to do with the ARM GCC version the Docker image currently uses, we're currently on 6.3.1, and I think 5.4.0? for AVR.

There are some PRs to update them:
https://github.com/qmk/qmk_base_container/pulls

**Comment by elagil** [comment]
> we're currently on 6.3.1

Well, I am also on that version: `gcc version 6.3.1 20170620 (15:6.3.1+svn253039-1build1)`

**Comment by elagil** [comment]
I tested the build with a docker VM and it failed equally. It seems like it uses different optimization or so, leading to more occupied RAM. Probably, the values from https://github.com/qmk/qmk_firmware/pull/10685 result in RAM usage that is sometimes too close to the limits of the processor.

**Comment by elagil** [comment]
Ran some tests and here are the results:

gcc version 6.3.1 20170620 (15:6.3.1+svn253039-1build1) on WSL compiles fine
gcc version 7.3.1 20180622 (release) [ARM/embedded-7-branch revision 261907] (15:7-2018-q2-6) on Debian 10 compiles fine

Only the docker fails, although it has gcc version 6.3.1. Is something different about the docker compiler?

**Pull request merged**